src/components/comment-input.component.js

```diff
@@ -1,51 +1,49 @@
 // @flow
 
 import React, { Component } from 'react';
-import {
-  StyleSheet,
-  View,
-  TouchableOpacity,
-  Text,
-  TextInput,
-  Platform,
-} from 'react-native';
+import { View, Text, Platform } from 'react-native';
 import { Icon } from 'react-native-elements';
+import styled from 'styled-components/native';
 
 import { MentionArea } from 'components';
 import { translate } from 'utils';
 import { colors, fonts, normalize } from 'config';
 
-const styles = StyleSheet.create({
-  container: {
-    borderTopColor: colors.greyLight,
-    borderTopWidth: 1,
-    backgroundColor: 'transparent',
-  },
-  wrapper: {
-    padding: 10,
-    flexDirection: 'row',
-    alignItems: 'center',
-    justifyContent: 'center',
-  },
-  textInput: {
-    fontSize: normalize(12),
-    flex: 1,
-    marginRight: 5,
-    color: colors.black,
-    ...fonts.fontPrimary,
-  },
-  postButtonContainer: {
-    flex: 0.15,
-    alignItems: 'flex-end',
-    justifyContent: 'center',
-  },
-  postButtonDisabled: {
-    color: colors.grey,
-  },
-  postButtonEnabled: {
-    color: colors.primaryDark,
-  },
-});
+const Container = styled.View`
+  border-top-color: ${colors.greyLight};
+  border-top-width: 1;
+  background-color: transparent;
+`;
+
+const Wrapper = styled.View`
+  flex-direction: row;
+  align-items: center;
+  padding: 10px;
+  justify-content: center;
+`;
+
+const InputText = styled.TextInput`
+  font-size: ${normalize(12)};
+  flex: 1;
+  margin-right: 5;
+  color: ${colors.black};
+  ${fonts.fontPrimary};
+`;
+const TextInputText = InputText.withComponent(Text);
+
+const PostButtonContainerCanPost = styled.TouchableOpacity`
+  flex: 0.15;
+  align-items: flex-end;
+  justify-content: center;
+`;
+const PostButtonContainerCantPost = PostButtonContainerCanPost.withComponent(
+  View
+);
+
+const PostButtonIcon = ({ style }) => <Icon name="send" iconStyle={style} />;
```

Comment:
I think you need to add props validation to this function.
What I think is that you need to do something like:
```js
type PostButtonIconProps = {
  style: Object
};

const PostButtonIcon = ({ style} : PostButtonIconProps)=> <Icon name="send" iconStyle={style} />;
```

`PostButtonIconProps` is a `flow` type annotation. 

**Welcome** to Gitpoint! Let us know if you have any other question!